collector/monitors/update_global_index_monitor.go

```diff
@@ -0,0 +1,236 @@
+		firstCheck = true
+	}
+
+	iterations := 0
+	var maxProcessedID int
+	var maxProcessedIDPerRequest int
+	var alreadyProcessedFound bool
+	m.InitMetrics()
+	for iterations < threshold {
+		p := transactions.GetV1TxsParams{}
+		p.SetAccount(&m.ContractAddress)
+		p.SetContext(ctx)
+		p.SetOffset(offset)
+
+		resp, err := m.apiClient.Transactions.GetV1Txs(&p)
+		if err != nil {
+			return fmt.Errorf("failed to fetch transaction history for UpdateGlobalIndexBotContract account: %w", err)
+		}
+
+		maxProcessedIDPerRequest, alreadyProcessedFound = m.processTransactions(resp.Payload.Txs, m.lastMaxCheckedID)
+		fetchedTxs += len(resp.Payload.Txs)
+		maxProcessedID = maxInt(maxProcessedID, maxProcessedIDPerRequest)
+		if alreadyProcessedFound || firstCheck {
+			break
+		}
+		offset = resp.Payload.Next
+		iterations++
+	}
+	m.lastMaxCheckedID = maxProcessedID
+	if threshold == iterations {
+		m.logger.Warning("update global index processing stopped due to requests threshold - ", threshold)
+	}
+	m.logger.Infoln("update global index txs fetched:", fetchedTxs)
+	m.logger.Infoln("update global index state:", m.metrics)
+	return nil
+}
+
+func maxInt(a, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func (m *UpdateGlobalIndexMonitor) processTransactions(txs []*models.GetTxListResultTxs, previousMaxCheckedID int) (newMaxCheckedID int, alreadyProcessedFound bool) {
+	for i, tx := range txs {
+		if i == 0 {
+			newMaxCheckedID = maxInt(int(*tx.ID), previousMaxCheckedID)
+		}
+		if previousMaxCheckedID == int(*tx.ID) {
+			// we have already checked this and earlier transactions
+			m.logger.Infoln("stopping processing, last checked transaction is found:", previousMaxCheckedID)
+			alreadyProcessedFound = true
+			break
+		}
+		switch isTxUpdateGlobalIndex(tx) {
+		case SuccessfulUpdateGlobalIndexTX:
+			m.metrics[UpdateGlobalIndexSuccessfulTxSinceLastCheck]++
+		case FailedUpdateGlobalIndexTx:
+			m.metrics[UpdateGlobalIndexFailedTxSinceLastCheck]++
+			m.logger.Warning("failed tx detected: ", getTxRawLog(tx))
+		case NonUpdateGlobalIndexTX:
+		}
+		m.metrics[UpdateGlobalIndexGasUsed] += gasUsed(m.logger, tx)
+		m.metrics[UpdateGlobalIndexGasWanted] += gasWanted(m.logger, tx)
+		m.metrics[UpdateGlobalIndexUUSDFee] += uusdFee(m.logger, tx)
+	}
+	return newMaxCheckedID, alreadyProcessedFound
+}
+
+func (m UpdateGlobalIndexMonitor) GetMetrics() map[Metric]float64 {
+	return m.metrics
+}
+
+func getTxRawLog(tx *models.GetTxListResultTxs) string {
+	if tx == nil || tx.RawLog == nil {
+		return ""
+	}
+	return *tx.RawLog
+}
+
+func isTxUpdateGlobalIndex(tx *models.GetTxListResultTxs) UpdateGlobalIndexTxsVariants {
+	if tx == nil || tx.Tx == nil || tx.Tx.Value == nil || len(tx.Tx.Value.Msg) == 0 {
+		return NonUpdateGlobalIndexTX
+	}
+	for _, msg := range tx.Tx.Value.Msg {
+		if msg.Value == nil || msg.Value.ExecuteMsg == nil {
+			continue
+		}
+		if *msg.Value.ExecuteMsg == UpdateGlobalIndexBase64Encoded && len(tx.Logs) > 0 {
```

Comment:
Is this the proper way to check for a failed transaction (by checking the logs)?

Reply:
i think so. i made my point in comment